core/src/apps/hedera/sign_tx.py

```diff
@@ -0,0 +1,134 @@
+from trezor.crypto.curve import ed25519
+from trezor.messages import HederaSignTx
+from trezor.messages import HederaSignedTx
+from trezor.strings import format_amount
+
+from apps.common import paths
+from apps.common.keychain import auto_keychain
+
+from .layout import (
+    confirm_account,
+    confirm_associate_token,
+    confirm_create_account,
+    confirm_transfer_hbar,
+    confirm_transfer_token,
+    format_account,
+)
+
+transaction_types = {
+    "CONFIRM": "Confirm Account",
+    "CREATE": "Create Account",
+    "TOKEN": "Token Transfer",
+    "HBAR": "Hbar Transfer",
+    "ASSOCIATE": "Token Associate",
+}
+
+
+def validate_tx(msg: HederaSignTx) -> str | None:
+    if msg.tx is not None:
+        # Create Account
+        if msg.tx.cryptoCreateAccount is not None:
+            # can't be both
+            if msg.tx.cryptoTransfer is not None:
+                return None
+
+            return transaction_types["CREATE"]
+
+        # Confirm Account, Transfer Hbar, Transfer Token
+        if msg.tx.cryptoTransfer is not None:
+            # Hbar transfer
+            if msg.tx.cryptoTransfer.transfers is not None:
+                # but also has token transfers (unsupported)
+                if msg.tx.cryptoTransfer.tokenTransfers is not None:
+                    return None
+
+                # at most two parties (confirm account, hbar transfer)
+                if len(msg.tx.cryptoTransfer.transfers.accountAmounts) > 2:
+                    return None
+
+                # between sender and one other account
+                if len(msg.tx.cryptoTransfer.transfers.accountAmounts) == 2:
+                    return transaction_types["HBAR"]
+
+                # transfer 0 hbar to nobody, with a max fee of 1 tinybar
+                # this fails precheck on hedera in a way that signifies whether or not the account
+                # used as the operator is associated with the key used to sign this transaction
+                if (
+                    len(msg.tx.cryptoTransfer.transfers.accountAmounts) == 1
+                    and msg.tx.cryptoTransfer.transfers.accountAmounts[0].amount == 0
```

Comment:
Missing check for `msg.tx.cryptoTransfer.transfers.accountAmounts[0].amount is not None`